tests/backend/test_postgres_db.py

```diff
@@ -158,3 +162,32 @@ async def test_retry_policy_allow_retry(postgresql_url, asyncio_loop):
 
             # Cancel the backend nursery
             nursery.cancel_scope.cancel()
+
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_datetime(postgresql_url, backend_factory):
+    now = DateTime.now()
```

Comment:
What the test tests:
- serialization with rust datetime
- deserialization with python datetime

what should also be tested:
- serialization with python datetime
- deserialization with rust datetime

I think the simplest way to do that is to open two connections at the same time
```python
async with triopg.connect(postgresql_url) as vanilla_conn:
    async with triopg.connect(postgresql_url) as custom_conn:
        await handle_datetime(custom_conn, "timestamptz")

        await vanilla_conn.execute(...)  # Insert datetime
        await patched_conn.execute(...)  # Insert datetime

        await vanilla_conn.execute(...)  # retrieve datetime inserted by vanilla
        await vanilla_conn.execute(...)  # retrieve datetime inserted by patched
        await patched_conn.execute(...)  # retrieve datetime inserted by vanilla
        await patched_conn.execute(...)  # retrieve datetime inserted by patched
```